cmake/Corrosion.cmake

```diff
@@ -246,6 +247,8 @@ function(_add_cargo_build)
     set(cargo_target_option "$<IF:${if_not_host_build_condition},--target=${_CORROSION_RUST_CARGO_TARGET},--target=${_CORROSION_RUST_CARGO_HOST_TARGET}>")
     set(target_artifact_dir "$<IF:${if_not_host_build_condition},${_CORROSION_RUST_CARGO_TARGET},${_CORROSION_RUST_CARGO_HOST_TARGET}>")
 
+    set(flags_genex "$<TARGET_PROPERTY:${target_name},${_CORR_PROP_FLAGS}>")
```

Comment:
I think it would be nice to also support generator expressions for the additional arguments. This is e.g. already done for the `features` genex. Basically wrapping this expression like `$<GENEX_EVAL:$<TARGET_PROPERTY:${target_name},INTERFACE_CORROSION_CARGO_FLAGS>>` should be sufficient.
But maybe you can take a look a the test for `features` and add a similar test with a Genex for the arbitrary flags.

Reply:
I've added that now, but I must admit I have no experience with genex. I've practically copied what was in features and then tried to simplify where I could.